tasks.py

```diff
@@ -552,20 +552,33 @@ def lint(ctx, env="container"):
 
 
 @task(help={
-    "name": "name of the kind cluster to test.",
-    "export": "where to export kind logs"
+    "name": "name of the kind cluster to test (only kind uses).",
+    "export": "where to export kind logs.",
+    "kubeconfig_path": "kubeconfig path. By default, load kubeconfig from kind.",
+    "system_namespace": "comma separated kubernetes system namespace list.",
+    "service_pod_port": "port number that service pods open.",
+    "skip_docker": "don't use docker command in BGP testing."
 })
-def e2etest(ctx, name="kind", export=None):
+def e2etest(ctx, name="kind", export=None, kubeconfig_path=None, system_namespace="kube-system,metallb-system", service_pod_port=80, skip_docker=False):
     """Run E2E tests against development cluster."""
-    validate_kind_version()
-    clusters = run("kind get clusters", hide=True).stdout.strip().splitlines()
-    if name in clusters:
-        kubeconfig = tempfile.NamedTemporaryFile()
-        run("kind export kubeconfig --name={} --kubeconfig={}".format(name, kubeconfig.name), pty=True, echo=True)
+    if skip_docker:
+        opt_skip_docker = "-skip-docker"
     else:
-        raise Exit(message="Unable to find cluster named: {}".format(name))
+        opt_skip_docker = ""
+
+    if kubeconfig_path is None:
+        validate_kind_version()
+        clusters = run("kind get clusters", hide=True).stdout.strip().splitlines()
+        if name in clusters:
+            kubeconfig = tempfile.NamedTemporaryFile()
+            kubeconfig_path = kubeconfig.name
```

Comment:
Following on from my comment above about the names of the kubeconfig parameter for this task. This could be:

`kubeconfig = kubeconfig_file.name`